src/Console/Command.php

```diff
@@ -5,36 +5,66 @@
 use Sokil\Mongo\Migrator\Config;
 use Sokil\Mongo\Migrator\Console\Exception\ConfigurationNotFound;
 use Sokil\Mongo\Migrator\Manager;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Yaml\Yaml;
 
 abstract class Command extends \Symfony\Component\Console\Command\Command
 {
+    /**
+     * @var Config
+     */
     private $config;
-    
+
     /**
      *
      * @var \Sokil\Mongo\Migrator\Manager
      */
     private $manager;
     
     const CONFIG_FILENAME = 'mongo-migrator';
-    
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @throws ConfigurationNotFound
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        // config file path
+        $configPath = $input->getOption('configuration');
+
+        if (!empty($configPath)) {
+            $this->config = new Config($this->readConfig($configPath));
+        }
+    }
+
     /**
-     *
      * @return \Sokil\Mongo\Migrator\Config
+     * @throws ConfigurationNotFound
      */
     protected function getConfig()
     {
         if (!$this->config) {
             $this->config = new Config($this->readConfig());
         }
-        
+
         return $this->config;
     }
-    
-    private function readConfig()
+
+    /**
+     * @param string $path
+     * @return mixed
+     * @throws ConfigurationNotFound
+     */
+    private function readConfig($path = '')
     {
-        $filename = $this->getProjectRoot() . '/' . self::CONFIG_FILENAME;
+        if (!empty($path) && file_exists($path)) {
+            $pathParts = pathinfo($path);
+            $filename = sprintf("%s/%s", $pathParts['dirname'], $pathParts['filename']);
+        } else {
+            $filename = sprintf("%s/%s", $this->getProjectRoot(), self::CONFIG_FILENAME);
```

Comment:
if passed filename not found, you deed to fail, this exception and die. If user pass some config it expects that this file will be used. If file not found, and you made migrations for example in other database, you will be a little wrong)

Reply:
okay, understand. Idea to move path initialisation above is better, I think